modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/routes/InstanceRoutes.scala

```diff
@@ -67,15 +74,17 @@ class InstanceRoutes(
           pathEndOrSingleSlash {
             (get & parameter('from.as[Int] ? pagination.from) & parameter('size.as[Int] ? pagination.size) & parameter('deprecated.as[Boolean].?)) { (from, size, deprecated) =>
               traceName("listInstancesOfDomain") {
-                queryBuilder.listingQuery(domainId, deprecated, None, Pagination(from, size)).response
+                val filter = Filter(deprecatedOrNoop(deprecated))
+                buildResponse(instanceQueries.list(domainId, filter, Pagination(from, size)), Pagination(from, size))
               }
             }
           } ~
           pathPrefix(Segment) { name =>
             pathEndOrSingleSlash {
               (get & parameter('from.as[Int] ? pagination.from) & parameter('size.as[Int] ? pagination.size) & parameter('deprecated.as[Boolean].?)) { (from, size, deprecated) =>
                 traceName("listInstancesOfSchemaName") {
-                  queryBuilder.listingQuery(domainId, name, deprecated, None, Pagination(from, size)).response
+                  val filter = Filter(deprecatedOrNoop(deprecated))
```

Comment:
extract this case class construction out of the method, so the construction of the case class is not duplicated.